foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/version_specific/MlTrainImageTagMigrationSteps.java

```diff
@@ -0,0 +1,57 @@
+package com.boozallen.aissemble.upgrade.migration.version_specific;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractMigrationTest;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import java.io.File;
+import java.nio.file.Path;
+
+public class MlTrainImageTagMigrationSteps extends AbstractMigrationTest {
+    public static final String MIGRATION_ROOT = "version-specific/MlTrainImageTagMigration/migration/ml-training/%svalues-dev.yaml";
+
+    @Given("a ML training pipeline values-dev file that does not have the imagePullPolicy and tag set")
+    public void a_ml_training_pipeline_values_dev_file_that_does_not_have_the_image_pull_policy_and_tag_set(){
+        testFile = getTestFile(String.format(MIGRATION_ROOT, ""));
+    }
+
+    @Given("a ML training pipeline values-dev file with no {string} defined")
+    public void a_values_dev_file_with_no_defined(String configuration) {
+        testFile = getTestFile(String.format(MIGRATION_ROOT, String.format("no-%s-", configuration)));
+    }
+
+    @Given("a ML training pipeline values-dev file that already has the imagePullPolicy set")
+    public void a_ml_training_pipeline_values_dev_file_that_already_has_the_image_pull_policy_set(){
+        testFile = getTestFile(Path.of("version-specific", "MlTrainImageTagMigration", "migration", "ml-training", "skip-values-dev.yaml").toString());
+    }
+
+    @When("the 1.13.0 ML train image tag migration executes")
+    public void the_ml_train_image_tag_migration_executes(){
+        MlTrainImageTagMigrationHelper migration = new MlTrainImageTagMigrationHelper();
+        performMigration(migration);
+    }
+
+    @Then("the image configuration is added")
+    public void the_image_configuration_is_added(){
+        String testFilePath = testFile.getPath().replaceAll("migration", "validation");
+        File validatedFile = new File(testFilePath);
+        assertLinesMatch("The image configuration was not added", validatedFile, testFile);
```

Comment:
S: Use abstract methods
```suggestion
        assertMigrationSuccess();
        assertTestFileMatchesExpectedFile("The image configuration was not added");
```